diff --git a/CMakeLists.txt b/CMakeLists.txt
index c63f22d..477610e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,4 +12,5 @@ find_package(SDL2_mixer REQUIRED)
 find_package(ZLIB REQUIRED)
 find_package(FluidSynth REQUIRED)
 
-add_subdirectory("${PROJECT_SOURCE_DIR}/src")
\ No newline at end of file
+add_subdirectory("${PROJECT_SOURCE_DIR}/src")
+install(TARGETS ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
