diff --git a/src/game/g_main.pas b/src/game/g_main.pas
index a485a25..b772cfd 100644
--- a/src/game/g_main.pas
+++ b/src/game/g_main.pas
@@ -270,8 +270,8 @@ var
     if not forceBinDir then
     begin
     {$IFDEF USE_SDL2}
-      AddDir(result, SDL_GetBasePath());
-      AddDir(result, SDL_GetPrefPath('', 'doom2df'));
+      AddDir(result, '/app/extra/doom2df');
+      AddDir(result, '/app/extra/doom2df/data');
     {$ENDIF}
     {$IFDEF WINDOWS}
       appdata := GetEnvironmentVariable('APPDATA') + '\doom2df';
@@ -279,11 +278,8 @@ var
         AddDir(result, appdata);
     {$ENDIF}
     {$IF DEFINED(UNIX) AND NOT DEFINED(DARWIN) AND NOT DEFINED(ANDROID)}
-      AddDir(result, '/usr/share/doom2df');
-      AddDir(result, '/usr/local/share/doom2df');
-      home := GetEnvironmentVariable('HOME');
-      if home <> '' then
-        AddDir(result, ConcatPaths([home, '.doom2df']));
+      AddDir(result, GetEnvironmentVariable('XDG_DATA_HOME'));
+      AddDir(result, '/app/extra/doom2df');
     {$ENDIF}
     {$IFDEF DARWIN}
       bundle := GetBundlePath();
@@ -330,7 +326,7 @@ var
     if not forceBinDir then
     begin
     {$IFDEF USE_SDL2}
-      AddDir(result, SDL_GetPrefPath('', 'doom2df'));
+      AddDir(result, GetEnvironmentVariable('XDG_DATA_HOME'));
     {$ENDIF}
     {$IFDEF WINDOWS}
       appdata := GetEnvironmentVariable('APPDATA') + '\doom2df';
@@ -338,9 +334,7 @@ var
         AddDir(result, appdata);
     {$ENDIF}
     {$IF DEFINED(UNIX) AND NOT DEFINED(DARWIN) AND NOT DEFINED(ANDROID)}
-      home := GetEnvironmentVariable('HOME');
-      if home <> '' then
-        AddDir(result, ConcatPaths([home, '.doom2df']));
+      AddDir(result, GetEnvironmentVariable('XDG_DATA_HOME'));
     {$ENDIF}
     {$IFDEF DARWIN}
       dirArr := NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, true);
