diff --git a/CMakeLists.txt b/CMakeLists.txt
index a505fc1..4d33bad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,24 +28,24 @@ endif()
 
 cmake_policy(SET CMP0135 NEW)
 
-include(FetchContent)
+if(MSVC)
+	include(FetchContent)
 
-FetchContent_Declare(
-	adlmidi
-	GIT_REPOSITORY https://github.com/Wohlstand/libADLMIDI.git
-	GIT_TAG v1.6.1
-	EXCLUDE_FROM_ALL
-)
-FetchContent_Declare(
-	inih
-	GIT_REPOSITORY https://github.com/benhoyt/inih.git
-	GIT_TAG r62
-	EXCLUDE_FROM_ALL
-)
+	FetchContent_Declare(
+		adlmidi
+		GIT_REPOSITORY https://github.com/Wohlstand/libADLMIDI.git
+		GIT_TAG v1.6.1
+		EXCLUDE_FROM_ALL
+	)
+	FetchContent_Declare(
+		inih
+		GIT_REPOSITORY https://github.com/benhoyt/inih.git
+		GIT_TAG r62
+		EXCLUDE_FROM_ALL
+	)
 
-FetchContent_MakeAvailable(adlmidi inih)
+	FetchContent_MakeAvailable(adlmidi inih)
 
-if(MSVC)
 	set(BARRETT_SDL2_VERSION "2.32.10")
 	FetchContent_Declare(
 		sdl2_msvc
