diff --git a/src/common/audio/sound/oalsound.cpp b/src/common/audio/sound/oalsound.cpp
index dd80f12e1..05fbc94f0 100644
--- a/src/common/audio/sound/oalsound.cpp
+++ b/src/common/audio/sound/oalsound.cpp
@@ -524,7 +524,7 @@ public:
 
 #define AREA_SOUND_RADIUS  (32.f)
 
-#define PITCH_MULT (0.7937005f) /* Approx. 4 semitones lower; what Nash suggested */
+#define PITCH_MULT (0.95f)
 
 static size_t GetChannelCount(ChannelConfig chans)
 {
@@ -1746,12 +1746,12 @@ void OpenALSoundRenderer::UpdateListener(SoundListener *listener)
 
 			if(EnvSlot != 0 && *snd_waterreverb)
 			{
-				// Find the "Underwater" reverb environment
-				env = S_FindEnvironment(0x1600);
+				// Find the "DSP Water" reverb environment
+				env = S_FindEnvironment(0xffff);
 				LoadReverb(env ? env : DefaultEnvironments[0]);
 
 				alFilterf(EnvFilters[0], AL_LOWPASS_GAIN, 1.f);
-				alFilterf(EnvFilters[0], AL_LOWPASS_GAINHF, 0.125f);
+				alFilterf(EnvFilters[0], AL_LOWPASS_GAINHF, 0.2f);
 				alFilterf(EnvFilters[1], AL_LOWPASS_GAIN, 1.f);
 				alFilterf(EnvFilters[1], AL_LOWPASS_GAINHF, 1.f);
 
diff --git a/src/common/audio/sound/s_environment.cpp b/src/common/audio/sound/s_environment.cpp
index 5b4ea79f1..f9249c6d0 100644
--- a/src/common/audio/sound/s_environment.cpp
+++ b/src/common/audio/sound/s_environment.cpp
@@ -128,14 +128,13 @@ static const char *BoolNames[3] = { "False", "True", NULL };
 
 static ReverbContainer DSPWater =
 {
-	// Based on the "off" reverb, this one uses the software water effect,
-	// which is completely independant from EAX-like reverb.
+	// copied from forest
 	NULL,
 	"DSP Water",
 	0xffff,
 	true,
 	false,
-	{0, 0,	7.5f,	1.00f, -10000, -10000, 0,   1.00f,  1.00f, 1.0f,  -2602, 0.007f, 0.0f,0.0f,0.0f,   200, 0.011f, 0.0f,0.0f,0.0f, 0.250f, 0.00f, 0.25f, 0.000f, -5.0f, 5000.0f, 250.0f, 0.0f,   0.0f,   0.0f, 0x33f },
+	{0,15,	38.0f,	0.30f, -1000,  -3300,  0,   1.49f,  0.54f, 1.0f,  -2560, 0.162f, 0.0f,0.0f,0.0f,  -229, 0.088f, 0.0f,0.0f,0.0f, 0.125f, 1.00f, 0.25f, 0.000f, -5.0f, 5000.0f, 250.0f, 0.0f,  79.0f, 100.0f, 0x3f },
 	true
 };
 
diff --git a/src/g_game.cpp b/src/g_game.cpp
index 4a72d4d24..1f9ad07a4 100644
--- a/src/g_game.cpp
+++ b/src/g_game.cpp
@@ -440,6 +450,7 @@ CCMD (invnext)
 			VMValue param = players[consoleplayer].mo;
 			VMCall(func, &param, 1, nullptr, 0);
 		}
+	    S_Sound(CHAN_AUTO, 0, "INVMOVE", 1.0, ATTN_NONE);
 	}
 }
 
@@ -452,6 +463,7 @@ CCMD(invprev)
 			VMValue param = players[consoleplayer].mo;
 			VMCall(func, &param, 1, nullptr, 0);
 		}
+	    S_Sound(CHAN_AUTO, 0, "INVMOVE", 1.0, ATTN_NONE);
 	}
 }
 
