diff --git a/src/d_buttons.h b/src/d_buttons.h
index 341ead790..aefe3af8b 100644
--- a/src/d_buttons.h
+++ b/src/d_buttons.h
@@ -51,1 +51,4 @@ enum
-	Button_MoveUp,
+	Button_MoveUp,
+   Button_SonataUp,
+   Button_SonataDown,
+   Button_SonataLeft,
diff --git a/src/d_event.h b/src/d_event.h
index 013e096d3..7b8320f50 100644
--- a/src/d_event.h
+++ b/src/d_event.h
@@ -72,4 +70,9 @@ typedef enum
 	BT_RUN			= 1<<25,
+
+	BT_SNUP	    	= 1<<26,
+	BT_SNDOWN		= 1<<27,
+	BT_SNLEFT		= 1<<28,
+
 } buttoncode_t;
 
 // Called by IO functions when input is detected.
diff --git a/src/d_main.cpp b/src/d_main.cpp
index 15c59342f..70e48d370 100644
--- a/src/d_main.cpp
+++ b/src/d_main.cpp
@@ -2932,1 +2932,7 @@ static const char *DoomButtons[] =
-	"moveup" };
+	"moveup",
+   "sonataup",
+   "sonatadown",
+   "sonataleft",
+   "sonataright",
+   "sonatause"
+};
diff --git a/src/g_game.cpp b/src/g_game.cpp
index 4a72d4d24..1f9ad07a4 100644
--- a/src/g_game.cpp
+++ b/src/g_game.cpp
@@ -669,2 +669,6 @@ void G_BuildTiccmd (ticcmd_t *cmd)
 	if (buttonMap.ButtonDown(Button_User4))			cmd->buttons |= BT_USER4;
 
+	if (buttonMap.ButtonDown(Button_SonataUp))		cmd->buttons |= BT_SNUP;
+	if (buttonMap.ButtonDown(Button_SonataDown))	cmd->buttons |= BT_SNDOWN;
+	if (buttonMap.ButtonDown(Button_SonataLeft))	cmd->buttons |= BT_SNLEFT;
+
diff --git a/wadsrc/static/zscript/constants.zs b/wadsrc/static/zscript/constants.zs
index aa1109de4..38a105959 100644
--- a/wadsrc/static/zscript/constants.zs
+++ b/wadsrc/static/zscript/constants.zs
@@ -882,1 +882,5 @@ enum EButtons
 	BT_RUN			= 1<<25,
+
+	BT_SNUP	    	= 1<<26,
+	BT_SNDOWN		= 1<<27,
+	BT_SNLEFT		= 1<<28,
