diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2bb65f338..0bddc3836 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7 +7 @@
-project(UZDoom)
+project(eos)
@@ -192,1 +192,1 @@ IF( NOT CMAKE_BUILD_TYPE )
-set( ZDOOM_EXE_NAME "uzdoom" CACHE FILEPATH "Name of the executable to create" )
+set( ZDOOM_EXE_NAME "eos" CACHE FILEPATH "Name of the executable to create" )
diff --git a/src/version.h b/src/version.h
index 1e7521ca9..f1ec040c1 100644
--- a/src/version.h
+++ b/src/version.h
@@ -44,1 +44,1 @@ const char *GetVersionString();
-#define VERSIONSTR "5.0.0"
+#define VERSIONSTR "5.0.0-eos"
@@ -75,1 +75,1 @@ const char *GetVersionString();
-#define GAMESIG "UZDOOM"
+#define GAMESIG "eos"
@@ -85,1 +85,1 @@ const char *GetVersionString();
-#define BASEWAD "uzdoom.pk3"
+#define BASEWAD "eos.pk3"
@@ -92,6 +92,6 @@ const char *GetVersionString();
-#define GAMENAME "UZDoom"
-#define WGAMENAME L"UZDoom"
-#define GAMENAMELOWERCASE "uzdoom"
-#define APPID "org.zdoom.UZDoom"
-#define QUERYIWADDEFAULT true
-#define BUGS_URL "https://github.com/UZDoom/UZDoom/issues"
+#define GAMENAME "eos"
+#define WGAMENAME L"eos"
+#define GAMENAMELOWERCASE "eos"
+#define APPID "horse.msx.eos"
+#define QUERYIWADDEFAULT false
+#define BUGS_URL "https://msx.horse/eos/"
diff --git a/src/win32/icon1.ico b/src/win32/icon1.ico
index 8238934c6..76731e74f 100644
Binary files a/src/win32/icon1.ico and b/src/win32/icon1.ico differ
diff --git a/src/win32/zdoom.rc b/src/win32/zdoom.rc
index 52bec74dc..130058a91 100644
--- a/src/win32/zdoom.rc
+++ b/src/win32/zdoom.rc
@@ -72,3 +72,3 @@ BEGIN
-    "            VALUE ""FileDescription"", ""UZDoom""\r\n"
+    "            VALUE ""FileDescription"", ""end of starchild""\r\n"
     "            VALUE ""FileVersion"", RC_FILEVERSION2\r\n"
-    "            VALUE ""InternalName"", ""UZDoom""\r\n"
+    "            VALUE ""InternalName"", ""end of starchild""\r\n"
@@ -74,5 +74,5 @@ BEGIN
-    "            VALUE ""OriginalFilename"", ""uzdoom.exe""\r\n"
-    "            VALUE ""ProductName"", ""UZDoom""\r\n"
+    "            VALUE ""OriginalFilename"", ""eos.exe""\r\n"
+    "            VALUE ""ProductName"", ""eos""\r\n"
     "            VALUE ""ProductVersion"", GIT_VERSION\r\n"
     "        END\r\n"
     "    END\r\n"
diff --git a/wadsrc/CMakeLists.txt b/wadsrc/CMakeLists.txt
index 05b3f8c18..3175b3b3d 100644
--- a/wadsrc/CMakeLists.txt
+++ b/wadsrc/CMakeLists.txt
@@ -1 +1 @@
-add_pk3(uzdoom.pk3 ${CMAKE_CURRENT_SOURCE_DIR}/static)
+add_pk3(eos.pk3 ${CMAKE_CURRENT_SOURCE_DIR}/static)
@@ -3 +3 @@
-generate_language_files(uzdoom_pk3 static/language.csv ENGINE)
+generate_language_files(eos_pk3 static/language.csv ENGINE)
@@ -15 +15 @@
-add_dependencies(uzdoom_pk3 copy_metadata)
+add_dependencies(eos_pk3 copy_metadata)
@@ -17 +17 @@
-	TARGET uzdoom_pk3
+	TARGET eos_pk3
diff --git a/src/g_game.cpp b/src/g_game.cpp
index 4a72d4d24..1f9ad07a4 100644
--- a/src/g_game.cpp
+++ b/src/g_game.cpp
@@ -2309,1 +2345,1 @@ void G_DoAutoSave ()
-	description.Format("Autosave %s", readableTime);
+	description.Format("%s - %s", readableTime, primaryLevel->LevelName.GetChars());
