diff --git a/src/quake2/src/unix/sys_unix.c b/src/quake2/src/unix/sys_unix.c
index 0c58b8b..ddabaee 100644
--- a/src/quake2/src/unix/sys_unix.c
+++ b/src/quake2/src/unix/sys_unix.c
@@ -366,1 +366,1 @@ char* Sys_GetHomeDir(void)
-	char* home = getenv("HOME");
+	char* home = getenv("XDG_DATA_HOME");
@@ -381,1 +381,1 @@ char* Sys_GetOSUserDir(void)
-	char* home = getenv("HOME");
+	char* home = getenv("XDG_DATA_HOME");
