diff --git a/CMakeLists.txt b/CMakeLists.txt
index a806283..d381e91 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,9 +6,8 @@ set(abuse_VERSION_MAJOR 0)
 set(abuse_VERSION_MINOR 9)
 set(abuse_VERSION_PATCH 0)
 set(abuse_VERSION_TWEAK 0)

 find_package(OpenGL REQUIRED)
-find_package(OpenCV REQUIRED)
 # Detect the platform, as there's quite a bit of platform variation
 
 if (WIN32)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7885110..576013a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -93,14 +93,6 @@ if(OPENGL_FOUND)
     target_link_libraries(abuse ${OPENGL_LIBRARIES})
 endif(OPENGL_FOUND)
 
-add_executable(abuse-tool
-	tool/abuse-tool.cpp
-	crc.cpp crc.h
-	tool/AR_SPEC.cpp tool/AR_SPEC.h
-	tool/AR_Help.cpp tool/AR_Help.h)
-
-target_link_libraries(abuse-tool imlib)
-
 include_directories(
     ${abuse_SOURCE_DIR}/src
     ${abuse_SOURCE_DIR}/src/lisp
@@ -133,7 +125,7 @@ if(WIN32)
     install(FILES ${SDL2MIXER_DLL_PATH}/SDL2_mixer.dll DESTINATION ".")
 else()
     # Under Linux, we want the tools in bin
-    install(TARGETS abuse abuse-tool RUNTIME DESTINATION bin
+    install(TARGETS abuse RUNTIME DESTINATION bin
     BUNDLE DESTINATION "${CMAKE_INSTALL_PREFIX}")
     if(APPLE)
         install(FILES "../osx/abuse.icns" DESTINATION "${CMAKE_INSTALL_PREFIX}/abuse.app/Contents/Resources")
