diff --git a/CMakeLists.txt b/CMakeLists.txt
index a806283..d381e91 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,8 @@ set(abuse_VERSION_MAJOR 0)
 set(abuse_VERSION_MINOR 9)
 set(abuse_VERSION_PATCH 0)
 set(abuse_VERSION_TWEAK 0)

+find_package(OpenGL REQUIRED)
 # Detect the platform, as there's quite a bit of platform variation
 
 if (WIN32)
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index cac258c..54981d4 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -1,7 +1,7 @@
 INSTALL(FILES
     abuse.lsp
     edit.lsp
-    hardness.lsp
+    user/hardness.lsp
 DESTINATION ${ASSETDIR})
 
 # No idea what this was for
@@ -327,7 +327,7 @@ INSTALL(FILES
 DESTINATION ${ASSETDIR}/addon/twist)
 
 INSTALL(FILES
-    defaults.prp
+    user/defaults.prp
 DESTINATION ${ASSETDIR})
 
 # Check if we have the non-free data
diff --git a/src/common.h b/src/common.h
index 1a14dd3..ee45256 100644
--- a/src/common.h
+++ b/src/common.h
@@ -14,7 +14,8 @@
 //
 // Globally required headers
 //
-#include <SDL_config.h>//AR (#include <SDL_config.h>)
+#include <SDL2/SDL_config.h>//AR (#include <SDL_config.h>)
+#include <stdint.h>
 #include <stdio.h>
 
 #ifdef _MSC_VER
diff --git a/src/crc.cpp b/src/crc.cpp
index c186364..d0b6f6f 100644
--- a/src/crc.cpp
+++ b/src/crc.cpp
@@ -12,6 +12,7 @@
 #   include "config.h"
 #endif
 
+#include <stdint.h>
 #include "crc.h"
 
 uint16_t calc_crc(void *buf, size_t len)
diff --git a/src/imlib/specs.cpp b/src/imlib/specs.cpp
index f242b29..fc820a9 100644
--- a/src/imlib/specs.cpp
+++ b/src/imlib/specs.cpp
@@ -14,7 +14,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <SDL_config.h>//AR (#include <SDL_config.h>)
+#include <SDL2/SDL_config.h>//AR (#include <SDL_config.h>)
 #include <ctype.h>
 #include <fcntl.h>
 #include <math.h>
diff --git a/src/imlib/specs.h b/src/imlib/specs.h
index 585afc0..20cffaa 100644
--- a/src/imlib/specs.h
+++ b/src/imlib/specs.h
@@ -14,7 +14,7 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <stdlib.h>
-#include <SDL_config.h>//AR (#include <SDL_config.h>)
+#include <SDL2/SDL_config.h>//AR (#include <SDL_config.h>)
 #include <string.h>
 
 #include "linked.h"
diff --git a/src/lisp/lisp.h b/src/lisp/lisp.h
index 791f045..38f2b46 100644
--- a/src/lisp/lisp.h
+++ b/src/lisp/lisp.h
@@ -12,7 +12,8 @@
 #define __LISP_HPP_
 
 #include <cstdlib>
-#include <SDL_config.h>//AR (#include <SDL_config.h>)
+#include <SDL2/SDL_config.h>//AR (#include <SDL_config.h>)
+#include <stdint.h>
 
 #ifdef L_PROFILE
 #include "timing.h"
diff --git a/src/sdlport/video.cpp b/src/sdlport/video.cpp
index 2c72135..b126044 100644
--- a/src/sdlport/video.cpp
+++ b/src/sdlport/video.cpp
@@ -27,7 +27,7 @@
 #ifdef WIN32
 #include <windows.h>
 #endif
-#include "Glee.h"
+#include "GL/glew.h"
 //Abuse linker->input:
 //	- opengl32.lib
 //	- GLee.lib
@@ -498,4 +498,4 @@ void update_window_done()
 
 	SDL_GL_SwapWindow(window);
 	//
-}
\ No newline at end of file
+}
diff --git a/src/specache.cpp b/src/specache.cpp
index e5c949c..3b20f54 100644
--- a/src/specache.cpp
+++ b/src/specache.cpp
@@ -12,6 +12,7 @@
 #   include "config.h"
 #endif
 
+#include <stdint.h>
 #include "specache.h"
 
 spec_directory_cache sd_cache;
