diff --git a/data/abuse.lsp b/data/abuse.lsp
index cc09aa9..c517635 100644
--- a/data/abuse.lsp
+++ b/data/abuse.lsp
@@ -9,4 +9,2 @@
-(if (not (load "lisp/english.lsp"))             ;; load language specific stuff
-    (progn
-      (print "Abuse data files not found. Maybe try the `-datadir' option?")
-      (quit)))
+(print "Loading English language")
+(load "lisp/english.lsp")
@@ -12,12 +12,14 @@
       (quit)))
 (setq load_warn T)
 
+(print "Loading base scripts...")
 (load "lisp/common.lsp")
 (load "lisp/userfuns.lsp")
 (load "lisp/options.lsp")
 (load "lisp/startup.lsp")
 (load "lisp/input.lsp")
 
+(print "Loading game scripts...")
 (load "lisp/sfx.lsp")
 (load "lisp/gates.lsp")
 (load "lisp/duong.lsp")
@@ -39,6 +41,7 @@
 
 ;; From the fRaBs addons. This is required in case we load a save file
 ;; from a fRaBs install, it will miss graphic tiles.
+(print "Loading fRaBs add-ons...")
 (load "addon/twist/f2chars.lsp")
 (load "addon/twist/f2ai.lsp")
 (load "addon/aliens/astartup.lsp")
@@ -48,6 +51,7 @@
 (load "addon/leon/4frabs.lsp")
 (load "addon/newart/newart.lsp")
 
+(print "Setting up game objects...")
 (setq bad_guy_list
   (list DARNEL ANT_ROOF TRACK_GUN SPRAY_GUN JUGGER ROB1 WHO ROCKET FLYER
         GREEN_FLYER BOSS_ANT DROID_JUGGER DROID DEATH_SKULL DEATH_UMBRELLA
@@ -59,8 +63,9 @@
         ANT_SHIP WALK_ROB T_REX FACE_HUGGER ALIEN_DRONE ALIEN_WARRIOR ANT
         ANT_JUMPER ANT_PRED ANT_GREATER2 WALK_ROB2 WALK_ROBHEAD))
 
+(print "Cleaning up...")
 (gc)              ;; garbage collection perm space
 (tmp-space)       ;; execute game code in tmp space which is not GC'ed
 
+(print "Running game.")
 (create_players DARNEL)
-
