diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index f9b99b6..2aa75d3 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -403,7 +402,7 @@ if(XXH_X86DISPATCH_ALLOW_AVX)
 endif()
 
 # Tracy
-if(NOT (CMAKE_BUILD_TYPE STREQUAL "Release"))
+if(NOT (CMAKE_BUILD_TYPE STREQUAL "Release") OR NOT (CMAKE_BUILD_TYPE STREQUAL "RelWithDbgInfo"))
 	option(TRACY_ENABLE_ON_CORE_COMPONENTS
 		"Enable and require Tracy to compile core components such as the renderer, shader recompiler and
 			HLE modules"
