diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 336243555..2cda139af 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -256,7 +256,6 @@ target_include_directories(vma INTERFACE
 	"${CMAKE_CURRENT_SOURCE_DIR}/VulkanMemoryAllocator-Hpp/include"
 	"${CMAKE_CURRENT_SOURCE_DIR}/VulkanMemoryAllocator-Hpp/VulkanMemoryAllocator/include")
 
-add_subdirectory(ffmpeg)

 
 if(APPLE OR NOT FORCE_BUILD_OPENSSL_MAC)
 	execute_process(
diff --git a/vita3k/CMakeLists.txt b/vita3k/CMakeLists.txt
index e254b4820..dee103e8a 100644
--- a/vita3k/CMakeLists.txt
+++ b/vita3k/CMakeLists.txt
@@ -70,6 +70,7 @@ file(WRITE ${CMAKE_BINARY_DIR}/vita3k/config/gitver.h
 )
 
 include_directories(${CMAKE_BINARY_DIR}/vita3k/config)
+include_directories(${CMAKE_SOURCE_DIR}/external/ffmpeg/include)
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 	set(LINUX TRUE)
diff --git a/vita3k/codec/CMakeLists.txt b/vita3k/codec/CMakeLists.txt
index b709aba7f..784c6b651 100644
--- a/vita3k/codec/CMakeLists.txt
+++ b/vita3k/codec/CMakeLists.txt
@@ -14,4 +14,13 @@ add_library(
 )
 
 target_include_directories(codec PUBLIC include)
-target_link_libraries(codec PRIVATE ffmpeg libatrac9 util) 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBAV REQUIRED IMPORTED_TARGET
+    libavfilter
+    libavformat
+    libavcodec
+    libswresample
+    libswscale
+    libavutil
+)
+target_link_libraries(codec PRIVATE PkgConfig::LIBAV libatrac9 util)
diff --git a/vita3k/ngs/CMakeLists.txt b/vita3k/ngs/CMakeLists.txt
index 9fc182a71..cc9a0515b 100644
--- a/vita3k/ngs/CMakeLists.txt
+++ b/vita3k/ngs/CMakeLists.txt
@@ -22,4 +22,13 @@ add_library(
 
 target_include_directories(ngs PUBLIC include)
 target_link_libraries(ngs PUBLIC codec)
-target_link_libraries(ngs PRIVATE util mem kernel cpu ffmpeg)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBAV REQUIRED IMPORTED_TARGET
+    libavfilter
+    libavformat
+    libavcodec
+    libswresample
+    libswscale
+    libavutil
+)
+target_link_libraries(ngs PRIVATE util mem kernel cpu PkgConfig::LIBAV)
diff --git a/vita3k/renderer/CMakeLists.txt b/vita3k/renderer/CMakeLists.txt
index a728caf7d..1f0af5720 100644
--- a/vita3k/renderer/CMakeLists.txt
+++ b/vita3k/renderer/CMakeLists.txt
@@ -47,7 +47,16 @@ add_library(
 
 target_include_directories(renderer PUBLIC include)
 target_link_libraries(renderer PUBLIC display mem stb shader glutil threads config util vkutil)
-target_link_libraries(renderer PRIVATE ddspp sdl2 stb ffmpeg xxHash::xxhash concurrentqueue)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBAV REQUIRED IMPORTED_TARGET
+    libavfilter
+    libavformat
+    libavcodec
+    libswresample
+    libswscale
+    libavutil
+)
+target_link_libraries(renderer PRIVATE ddspp sdl2 stb PkgConfig::LIBAV xxHash::xxhash concurrentqueue)
 
 # Marshmallow Tracy linking
 if(TRACY_ENABLE_ON_CORE_COMPONENTS)
