diff --git a/Quake/sys_sdl_unix.c b/Quake/sys_sdl_unix.c
index 58f7316b..26aab52a 100644
--- a/Quake/sys_sdl_unix.c
+++ b/Quake/sys_sdl_unix.c
@@ -272,17 +272,7 @@ static void Sys_GetUserdir (char *dst, size_t dstsize)
 {
 	size_t		n;
 	const char	*home_dir = NULL;
-	struct passwd	*pwent;
-
-	pwent = getpwuid( getuid() );
-	if (pwent == NULL)
-		perror("getpwuid");
-	else
-		home_dir = pwent->pw_dir;
-	if (home_dir == NULL)
-		home_dir = getenv("HOME");
-	if (home_dir == NULL)
-		Sys_Error ("Couldn't determine userspace directory");
+	home_dir = getenv("XDG_DATA_HOME");
 
 /* what would be a maximum path for a file in the user's directory...
  * $HOME/SYS_USERDIR/game_dir/dirname1/dirname2/dirname3/filename.ext
diff --git a/Quake/host_cmd.c b/Quake/host_cmd.c
index 58f7316b..26aab52a 100644
--- a/Quake/host_cmd.c
+++ b/Quake/host_cmd.c
@@ -21,6 +21,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 */
 
+#include "common.h"
 #include "quakedef.h"
 #include "q_ctype.h"
 #include "json.h"
@@ -1405,7 +1406,7 @@ void SaveList_Init (void)
 		FileList_Add (savename, &savelist);
 	}
 
-	if ((size_t) q_snprintf (dirname, sizeof (dirname), "%s/autosave", com_gamedir) < sizeof (dirname))
+	if ((size_t) q_snprintf (dirname, sizeof (dirname), "%s/autosave/%s", host_parms->userdir, com_gamedir) < sizeof (dirname))
 	{
 		for (find = Sys_FindFirst (dirname, "sav"); find; find = Sys_FindNext (find))
 		{
@@ -1413,7 +1414,7 @@ void SaveList_Init (void)
 			if (find->attribs & FA_DIRECTORY)
 				continue;
 			COM_StripExtension (find->name, filename, sizeof (filename));
-			if ((size_t) q_snprintf (savename, sizeof (savename), "autosave/%s", filename) < sizeof (savename))
+			if ((size_t) q_snprintf (savename, sizeof (savename), "%s/autosave/%s/%s", host_parms->userdir, com_gamedir, filename) < sizeof (savename))
 				FileList_Add (savename, &savelist);
 		}
 	}
diff --git a/Quake/menu.c b/Quake/menu.c
index 58f7316b..26aab52a 100644
--- a/Quake/menu.c
+++ b/Quake/menu.c
@@ -2066,7 +2066,7 @@ void M_Menu_Skill_f (void)
 
 	m_skill_canresume = false;
 	m_skill_lastplayed = 0;
-	q_snprintf (autosave, sizeof (autosave), "%s/autosave/%s.sav", com_gamedir, m_skill_mapname);
+	q_snprintf (autosave, sizeof (autosave), "%s/autosave/%s/%s.sav", host_parms->userdir, com_gamedir, m_skill_mapname);
 	if (Sys_FileExists (autosave))
 	{
 		time_t now, lastplayed;
