diff --git a/premake5.lua b/premake5.lua
index 6625475..6337d7d 100644
--- a/premake5.lua
+++ b/premake5.lua
@@ -119,7 +119,6 @@ filter("platforms:Linux")
   })
   pkg_config.all("gtk+-x11-3.0")
   links({
-    "stdc++fs",
     "dl",
     "lz4",
     "pthread",
@@ -153,6 +152,7 @@ filter({"platforms:Linux", "language:C++", "toolset:clang"})
   disablewarnings({
     "deprecated-register",
     "deprecated-volatile",
+    "unused-result",
     "switch",
     "deprecated-enum-enum-conversion",
     "attributes",
