diff --git a/premake5.lua b/premake5.lua
index 548140f..8728421 100644
--- a/premake5.lua
+++ b/premake5.lua
@@ -124,7 +124,6 @@ filter("platforms:Linux")
   --})
   pkg_config.all("gtk+-x11-3.0")
   links({
-    "stdc++fs",
     "dl",
     "lz4",
     "pthread",
@@ -137,6 +136,7 @@ filter({"platforms:Linux", "kind:*App"})
 filter({"language:C++", "toolset:clang or gcc"}) -- "platforms:Linux"
   disablewarnings({
     "switch",
+    "unused-result",
     "attributes",
   })
 