diff --git a/CMakeLists.txt b/CMakeLists.txt
index 115ffab45..d3a50a796 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -309,3 +309,9 @@ set(CMAKE_FOLDER "")
 
 # Set default startup project for Visual Studio
 set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT xenia-app)
+
+# Generate version.h
+include(cmake/version_h.cmake)
+generate_version_h()
+
+include(cmake/install.cmake)
