diff --git a/CMakeLists.txt b/CMakeLists.txt
index 193f6d8..386ea4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -249,7 +249,7 @@ else()
     add_compile_options(${GTK3_CFLAGS_OTHER})
 
     # Global Linux libraries
-    link_libraries(stdc++fs dl lz4 pthread rt)
+    link_libraries(dl lz4 pthread rt)
 
     # Link groups for executables (resolves circular deps)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--start-group")
