diff --git a/CMakeLists.txt b/CMakeLists.txt
index 056bd99db..007dcd08d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,35 +221,42 @@ include_directories(
 		"${PROJECT_SOURCE_DIR}/idlib"
 )

-find_package(ZLIB REQUIRED CONFIG)
-find_package(minizip REQUIRED CONFIG)
-find_package(CURL REQUIRED CONFIG)
+include(FindPkgConfig)
+# find_package(ZLIB REQUIRED CONFIG)
+# find_package(minizip REQUIRED CONFIG)
+# find_package(CURL REQUIRED CONFIG)
 find_package(OpenAL REQUIRED CONFIG)
 find_package(Ogg REQUIRED CONFIG)
-find_package(Vorbis REQUIRED CONFIG)
-find_package(JPEG REQUIRED CONFIG)
-find_package(PNG REQUIRED CONFIG)
+# find_package(Vorbis REQUIRED CONFIG)
+# find_package(JPEG REQUIRED CONFIG)
+# find_package(PNG REQUIRED CONFIG)
 find_package(ffmpeg REQUIRED CONFIG)
 find_package(pugixml REQUIRED CONFIG)
 find_package(MbedTLS REQUIRED CONFIG)
 find_package(glfw3 REQUIRED CONFIG)
 find_package(doctest REQUIRED CONFIG)
 find_package(tracy REQUIRED CONFIG)
+pkg_check_modules(ZLIB zlib REQUIRED)
+pkg_check_modules(minizip minizip REQUIRED)
+pkg_check_modules(CURL libcurl REQUIRED)
+pkg_check_modules(Vorbis vorbis REQUIRED)
+pkg_check_modules(JPEG libjpeg REQUIRED)
+pkg_check_modules(PNG libpng REQUIRED)
 
 target_link_libraries(TheDarkMod
-		ZLIB::ZLIB
-		minizip::minizip
-		CURL::libcurl
+		ZLIB
+		minizip
+		libcurl
 		OpenAL::OpenAL
-		vorbis::vorbis
+		vorbis
 		Ogg::ogg
-		JPEG::JPEG
-		PNG::PNG
-		ffmpeg::ffmpeg
+		libjpeg
+		libpng
+		ffmpeg
 		glfw
 		pugixml::pugixml
 		doctest::doctest
-		tracy::tracy
+		tracy
 )
 
 if (WIN32)
