diff --git a/CMakeLists.txt b/CMakeLists.txt
index 056bd99db..d174de5c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -221,35 +221,44 @@ include_directories(
 		"${PROJECT_SOURCE_DIR}/idlib"
 )

-find_package(ZLIB REQUIRED CONFIG)
+add_library(ZLIB::ZLIB INTERFACE IMPORTED)
-find_package(minizip REQUIRED CONFIG)
+add_library(minizip INTERFACE IMPORTED)
-find_package(CURL REQUIRED CONFIG)
+add_library(CURL INTERFACE IMPORTED)
 find_package(OpenAL REQUIRED CONFIG)
 find_package(Ogg REQUIRED CONFIG)
-find_package(Vorbis REQUIRED CONFIG)
-find_package(JPEG REQUIRED CONFIG)
-find_package(PNG REQUIRED CONFIG)
-find_package(ffmpeg REQUIRED CONFIG)
+add_library(Vorbis INTERFACE IMPORTED)
+add_library(JPEG INTERFACE IMPORTED)
+add_library(PNG INTERFACE IMPORTED)
 find_package(pugixml REQUIRED CONFIG)
 find_package(MbedTLS REQUIRED CONFIG)
 find_package(glfw3 REQUIRED CONFIG)
-find_package(doctest REQUIRED CONFIG)
-find_package(tracy REQUIRED CONFIG)
+# find_package(doctest REQUIRED CONFIG)
+# find_package(tracy REQUIRED CONFIG)
+pkg_check_modules(LIBAV REQUIRED IMPORTED_TARGET
+    libavdevice
+    libavfilter
+    libavformat
+    libavcodec
+    libswresample
+    libswscale
+    libavutil
+)

 target_link_libraries(TheDarkMod
 		ZLIB::ZLIB
-		minizip::minizip
-		CURL::libcurl
+		minizip
+		CURL
 		OpenAL::OpenAL
-		vorbis::vorbis
+		vorbis
 		Ogg::ogg
-		JPEG::JPEG
-		PNG::PNG
-		ffmpeg::ffmpeg
+		JPEG
+		PNG
+		PkgConfig::LIBAV
 		glfw
 		pugixml::pugixml
-		doctest::doctest
-		tracy::tracy
+		# doctest::doctest
+		# tracy::tracy
 )

 if (WIN32)
