diff --git a/sys/cmake/PrecompiledHeader.cmake b/sys/cmake/PrecompiledHeader.cmake
index baaf0df03..11d5a0c7e 100644
--- a/sys/cmake/PrecompiledHeader.cmake
+++ b/sys/cmake/PrecompiledHeader.cmake
@@ -70,9 +70,9 @@ function(export_all_flags Target Filename)
 	string(TOUPPER "CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}" _current_build_compile_flags)
 	set(_compile_flags "$<TARGET_PROPERTY:${Target},COMPILE_FLAGS>" "${CMAKE_CXX_FLAGS}" "${${_current_build_compile_flags}}" "-std=c++14")
 	set(_compile_options "$<TARGET_PROPERTY:${Target},COMPILE_OPTIONS>")
-	set(_include_directories "$<$<BOOL:${_include_directories}>:-I$<JOIN:${_include_directories},\n-I>\n>")
-	set(_compile_definitions "$<$<BOOL:${_compile_definitions}>:-D$<JOIN:${_compile_definitions},\n-D>\n>")
-	set(_compile_flags "$<$<BOOL:${_compile_flags}>:$<JOIN:${_compile_flags},\n>\n>")
-	set(_compile_options "$<$<BOOL:${_compile_options}>:$<JOIN:${_compile_options},\n>\n>")
+	set(_include_directories "-I$<JOIN:${_include_directories},\n-I>\n>")
+	set(_compile_definitions "-D$<JOIN:${_compile_definitions},\n-D>\n>")
+	set(_compile_flags "$<JOIN:${_compile_flags},\n>\n>")
+	set(_compile_options "$<JOIN:${_compile_options},\n>\n>")
 	file(GENERATE OUTPUT "${Filename}" CONTENT "${_compile_definitions}${_include_directories}${_compile_flags}${_compile_options}\n")
 endfunction()
