diff --git a/src/mainwindow.ui b/src/mainwindow.ui
index 07f23af..d896095 100644
--- a/src/mainwindow.ui
+++ b/src/mainwindow.ui
@@ -1294,7 +1294,6 @@ font: 13pt &quot;Arial&quot;;</string>
     <addaction name="actionOpenSettings"/>
     <addaction name="actionAbout"/>
     <addaction name="actionGithubDsdalauncher"/>
-    <addaction name="actionCheckForUpdatesDsdalauncher"/>
     <addaction name="actionTips"/>
    </widget>
    <widget class="QMenu" name="menudsda_doom">
@@ -1302,7 +1301,6 @@ font: 13pt &quot;Arial&quot;;</string>
      <string>dsda-doom</string>
     </property>
     <addaction name="actionGithubDsdadoom"/>
-    <addaction name="actionCheckForUpdatesDsdadoom"/>
    </widget>
    <widget class="QMenu" name="menuAdvanced">
     <property name="title">
diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index e43ef88..24b3125 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -91,10 +91,5 @@ MainWindow::~MainWindow() { delete ui; }
 
 void MainWindow::CheckUpdates()
 {
-    if (settings->value("updaterStartLauncher").toBool())
-    {
-        if (updateLauncherDialog(false)) return;
-    }
-    if (settings->value("updaterStartGame").toBool()) updateGameDialog(false);
 }
 
diff --git a/src/settings.ui b/src/settings.ui
index f5fe141..94d5cc5 100644
--- a/src/settings.ui
+++ b/src/settings.ui
@@ -998,7 +998,7 @@ border-radius:3px</string>
      </property>
     </widget>
    </widget>
-   <widget class="QWidget" name="tab_3">
+<!--   <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Updates</string>
     </attribute>
@@ -1066,7 +1066,7 @@ border-radius:3px</string>
        </property>
       </widget>
     </widget>
-   </widget>
+   </widget> -->
   </widget>
  </widget>
  <resources/>
diff --git a/src/settings.cpp b/src/settings.cpp
index e60ffb0..e16ee8d 100644
--- a/src/settings.cpp
+++ b/src/settings.cpp
@@ -166,12 +166,6 @@ void Settings::ReadSettings()
         ui->maxHistory_lineEdit->setText("20");
     }
 
-    if (!settings->value("updaterStartLauncher").isNull()) ui->updaterStartLauncher_checkBox->setChecked(settings->value("updaterStartLauncher").toBool());
-    else settings->setValue("updaterStartLauncher", true);
-
-    if (!settings->value("updaterStartGame").isNull()) ui->updaterStartGame_checkBox->setChecked(settings->value("updaterStartGame").toBool());
-    else settings->setValue("updaterStartGame", true);
-
     settings->sync();
 }
 
