diff --git a/src/mainwindow.ui b/src/mainwindow.ui
index 8ed1573..3a6c91f 100644
--- a/src/mainwindow.ui
+++ b/src/mainwindow.ui
@@ -7,18 +7,18 @@
     <x>0</x>
     <y>0</y>
     <width>350</width>
-    <height>475</height>
+    <height>500</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>350</width>
-    <height>475</height>
+    <height>500</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>350</width>
-    <height>475</height>
+    <height>500</height>
    </size>
   </property>
   <property name="windowTitle">
@@ -349,7 +350,7 @@ border-radius:3px</string>
       <x>20</x>
       <y>10</y>
       <width>311</width>
-      <height>151</height>
+      <height>128</height>
      </rect>
     </property>
     <property name="styleSheet">
