diff --git a/src/libtrx/engine/image.c b/src/libtrx/engine/image.c
index e907874d..4dac601f 100644
--- a/src/libtrx/engine/image.c
+++ b/src/libtrx/engine/image.c
@@ -374,7 +374,11 @@ bool Image_SaveToFile(const IMAGE *const image, const char *const path)
         goto cleanup;
     }
 
-    fp = File_Open(path, FILE_OPEN_WRITE);
+    char *xdg_path = Memory_Alloc(1024);
+    strcat(xdg_path, getenv("XDG_DATA_HOME"));
+    strcat(xdg_path, "/");
+    strcat(xdg_path, path);
+    fp = File_Open(xdg_path, FILE_OPEN_WRITE);
     if (fp == NULL) {
         LOG_ERROR("Cannot create image file: %s", path);
         goto cleanup;
@@ -468,6 +472,7 @@ bool Image_SaveToFile(const IMAGE *const image, const char *const path)
     }
 
 cleanup:
+    Memory_FreePointer(&xdg_path);
     if (error_code) {
         LOG_ERROR(
             "Error while saving image %s: %s", path, av_err2str(error_code));
